lib/auth/accesslist/service_test.go

```diff
@@ -0,0 +1,256 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package accesslist
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/stretchr/testify/require"
+
+	conv "github.com/gravitational/teleport/api/convert/teleport/accesslist/v1"
+	accesslistv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/accesslist/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/types/header"
+	"github.com/gravitational/teleport/lib/authz"
+	"github.com/gravitational/teleport/lib/backend/memory"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/services/local"
+	"github.com/gravitational/teleport/lib/tlsca"
+)
+
+func TestAccessLists(t *testing.T) {
```

Comment:
I would suggest separate tests for each RPC, possibly with the exception of Get (which tends to be necessary for other tests anyway).

Reply:
I'd like to push back on this if you don't mind -- it's convenient to use all of the various calls here to verify one another. However, if you feel strongly I can adjust.

Reply:
I think it's too much, even though Go tests tend to go overboard. If we can commit to not keep adding to it, then I suppose it's alright.

Reply:
It's all good -- I'll split this up here.